realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -281,6 +283,35 @@ public void refresh() {
     }
 
     /**
+     * Blocks the current thread util transactions are committed by other threads to this Realm or
+     * {@link #stopWaitForChange()} gets called.
+     *
+     * @return {@code true} if transactions are committed to this Realm, {@code false} if
+     * {@link #stopWaitForChange()} gets called.
+     * @throws IllegalStateException if attempting to wait within a transaction.
+     */
+    public boolean waitForChange() {
```

Comment:
Now `Realm` cannot wait in a thread with Looper. `IllegalStateException` is thrown.